scripts/benchmark-runner.py

```diff
@@ -389,6 +384,19 @@ def upload_results(output_dir, run_id):
     conbench_upload_callable()
 
 
+def upload(args):
+    try:
+        upload_results(
+            output_dir=args.output_dir,
+            run_id=args.run_id,
+            sha=args.sha,
+            pr_number=args.pr_number,
+        )
+    except Exception:
+        print("ERROR caught during uploading results:")
+        print(traceback.format_exc())
```

Comment:
This should not block the progress of this PR, but is important to think and talk about in general.

I didn't dive into the implementation of `upload_results()`. Its behavior w.r.t. error handling is hard to see through from here.

But here is how I would expect it to behave:

1. Since it consumes an HTTP API it should have deliberate retrying functionality. In a CI job it might try for a long-ish time to POST results before it fails, auto-healing mini outages of the receiving end. 

2. If an HTTP request fails there should be nice log output showing the details (error response contents, good error msgs, etc).

3. If uploading finally fails then we want this wrapper program here to exit with a non-zero status code (right?).

Would appreciate a quick comment on each of those three points.

When I see

```
    try:
...
    except Exception:
        print("ERROR caught during uploading results:")
        print(traceback.format_exc())
```
then this actually triggers my "oh, that's probably the number one group of bugs that I had to hunt over the years" -- re-writing an error without a _clean, explicit_ intent :-). The intent _might_ actually be that we do _not_ want to crash the interpreter upon _any_ exception happening during  `upload_results()`. But, and this is about point (3) above -- we want to exit non-zero in this case, right? Without the exception handler we would certainly get an interpreter crash including precisely `traceback.format_exc()`.

Reply:
This was left over from the CircleCI implementation where running and uploading benchmarks all happened during one job. We basically always wanted that build to be green. If benchmarks were regressive, we'd print a comment and exit 0. If uploading failed, we'd print this comment and exit 0. This was so the benchmark job wouldn't block PRs. (For the last few months, uploading always failed because the credentials were revoked.)

It's probably true that we want the `upload` job to fail on GHA, because it won't block PRs, and it will be easier to notice when this job failed, so we can fix it.

Reply:
As @austin3dickey says I really just refactored the existing code into a separate command without any changes. I agree that some changes/refactoring would be nice or are needed with the new format  of the workflow (just make it a proper typr cli package with tests...).

I do enjoy your long-form commentary though :nerd_face: